src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/SqlInternalConnectionTds.cs

```diff
@@ -808,6 +809,16 @@ protected override bool UnbindOnTransactionCompletion
             }
         }
 
+        /// <summary>
+        /// Validates if federated authentication is used, Access Token used by this connection is active for the next 10 minutes.
+        /// </summary>
+        internal override bool IsAccessTokenExpired
+        {
+            get
+            {
+                return _federatedAuthenticationInfoRequested && DateTime.FromFileTimeUtc(_fedAuthToken.expirationFileTime) < DateTime.UtcNow.AddMinutes(10d);
```

Comment:
We chatted offline, can we change 10 minutes to Connection Timeout instead? 10 minutes is too long :)

Reply:
What do you recommend as ceiling value for cases where users set Connection Timeout in hours?

Reply:
As discussed offline, ceiling set to 10 mins, buffer defaults to Connection Timeout.